api/serializers.py

```diff
@@ -28,6 +30,62 @@ def validate(self, attrs):
         return attrs
 
 
+class CoreModelWithDurationSerializer(CoreModelSerializer):
+    """
+    Specific serializer base for models with a "start" and "end" field.
+    """
+    child = serializers.PrimaryKeyRelatedField(
+        allow_null=True, allow_empty=True, queryset=models.Child.objects.all(),
+        required=False)
+
+    class Meta:
+        abstract = True
+        extra_kwargs = {
+            'start': {'required': False},
+            'end': {'required': False},
+        }
+
+    def validate(self, attrs):
+        # Check for a special "timer" data argument that can be used in place
+        # of "start" and "end" fields as well as "child" if it is set on the
+        # Timer entry.
+        timer = None
+        if 'timer' in self.initial_data:
+            try:
+                timer = models.Timer.objects.get(pk=self.initial_data['timer'])
+            except models.Timer.DoesNotExist:
+                raise ValidationError({'timer': ['Timer does not exist.']})
+            if timer.end:
+                end = timer.end
+            else:
+                end = timezone.now()
+            if 'child' not in attrs and timer.child:
```

Comment:
Potentially this should not check for `child` in `attrs` and should just hammer over it as `start` and `end` do (it the Timer has a related Child).